### PR TITLE
Add sklearn 1.5 to CI matrix

### PR DESCRIPTION
**Pull request by Alexsandruss**
### Description

Returns https://github.com/intel/scikit-learn-intelex/pull/1842.

Changes:
- Add sklearn 1.5 (py312) to CI matrix
- Change py38->py39 in CI matrix
- Disable dpctl and dpnp for py39 sklearn 1.0 job due to process hanging during tests
- Deselect two PCA tests related to "auto" solver check
- Update LogReg parameter
- Fix for global patch

Checks:
- [ ] I have reviewed my changes thoroughly before submitting this pull request.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have updated the documentation to reflect the changes, if necessary (updated in # - _add PR number_)
- [ ] The unit tests pass successfully.
- [ ] I have run it locally and tested the changes extensively.
- [ ] I have resolved any merge conflicts that might occur with the base branch.
- [ ] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/intel/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_
- [ ] I have added a respective label(s) to PR if I have a permission for that.

### CODE REVIEW

**Comment by Alexsandruss** [comment]
@mergifyio rebase

**Review by ethanglaser — APPROVED** [review]

File: deselected_tests.yaml [hunk 1]
```diff
@@ -29,6 +29,11 @@ deselected_tests:
   - neighbors/tests/test_neighbors.py::test_kneighbors_brute_backend[float64-kulsinski] <1.3
   - neighbors/tests/test_neighbors.py::test_radius_neighbors_brute_backend[kulsinski] <1.3
 
+  # sklearnex PCA always chooses "covariance_eigh" solver instead of "full" when solver="auto"
+  # resulting in solver assignment check failure for sklearn version >= 1.5
+  - decomposition/tests/test_pca.py::test_pca_svd_solver_auto[1000-500-400-full] >=1.5
+  - decomposition/tests/test_pca.py::test_pca_svd_solver_auto[1000-500-0.5-full] >=1.5
```
Comment (samir-nasibli) [inline deselected_tests.yaml:35]: Please cover this with ticket(s)

**Review by samir-nasibli — APPROVED** [review]
Thank you! Assuming green CI
